### ISSUE TITLE
[return-await] breaks with functions nested in try blocks

### ISSUE THREAD

**Issue by MichaelBelousov** [typescript-eslint/typescript-eslint#3600]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/return-await": "error",
  }
}
```

```TS
async function test() {
  try {
    const anyCallback = () => {};
    return Promise.resolve(5); // not considered an error
  } finally {
    return Promise.resolve(2);
  }
```

**Expected Result**

I expect this rule to consider the comment-marked line an error, because a promise is returned without 
being first awaited

**Actual Result**

But it isn't considered an error.
No errors are generated while linting.

**PR**

I submitted a fix PR (#3601). It uses a stack during AST
traversal to allow nested functions to be handled correctly.

Please let me know if there is anything I missed for contributing.
